nixos/modules/tasks/network-interfaces.nix

```diff
@@ -612,6 +657,93 @@ in
       };
     };
 
+    networking.wlanInterfaces = mkOption {
+      default = { };
+      example = {
+        wlan-adhoc0 = {
+          base = "wlp6s0";
+          type = "ibss";
+          mac = "02:00:00:00:00:01";
+        };
+      };
+      description =
+        ''
+          This option allows you to define vlan devices that tag packets
+          on top of a physical interface. The value of this option is an
+          attribute set. Each attribute specifies a vlan, with the name
+          specifying the name of the vlan interface.
```

Comment:
True, thanks! But as there is no real distinction between `virtual` and `other/default` WLAN interfaces, I would rather adapt the description. If needed, then I would rather call the single physical WLAN interface a `device` to separate it from the WLAN `interfaces`. Maybe not mentioning `virtual.../vlan/v...` at all is the best abstraction from a user's perspective?